app/features/build_runner/build_runner.rb

```diff
@@ -200,6 +202,19 @@ def setup_build_specific_environment_variables
       env_mapping[:CI_BRANCH] = env_mapping[:GIT_BRANCH]
       # env_mapping[:CI_PULL_REQUEST] = nil # TODO: do we have the PR information here?
 
+      # Now that we have the CI specific ENV variables, let's go through the ENV variables
+      # the user defined in their configuration
+      Services.environment_variable_service.environment_variables.each do |environment_variable|
```

Comment:
This is so awesome, we can use the infrastructure we already build to inject `BUILD_ID`, `CI_BRANCH` and similar ENV variables